### PR TITLE
Home

### PR DESCRIPTION
**Pull request by dev-ashishsethi**
https://user-images.githubusercontent.com/89299065/166146657-8817609f-af53-412e-a02e-2497e549a1b0.mp4

Features Added:

1. Add a note
2. change color
3. edit note
4. add labels 
5. add priority
6. filters
7. Log in

image
image

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,41 +1,27 @@
...
-            rel="noreferrer"
-          >
-            Contribute
-          </a>
-        </div>
-      </header>
+    <div className="layout-container">
+      {/* <> */}
+      {/* <LandingPage /> */}
+      {login && <Nav />}
```
Comment (PuneetP16) [inline src/App.js:17]: remove commented code

File: src/Components/Cards/Cards.jsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        data: { note: note },
+        headers: { authorization: token },
+      });
+      setOthersData(
+        response.data.notes.filter((note) => note.pinned === false)
+      );
+      console.log("cards", response.data.notes);
+    })();
+
+    console.log("in color handler");
```
Comment (PuneetP16) [inline src/Components/Cards/Cards.jsx:44]: remove console.log's

File: src/Components/Cards/Cards.jsx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    // setIsEditable(true);
+  };
+  const archiveHandler = () => {};
+  const trashHandler = () => {};
+
```
Comment (PuneetP16) [inline src/Components/Cards/Cards.jsx:49]: remove this handler, if not used.

File: src/Components/Cards/Cards.jsx [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+  return (
+    <div className={`card card-align-item-start ${data.singleColor}`}>
+      <h4 className={`title ${data.singleColor}`}>{data.title}</h4>
+
+      <p className={`description ${data.singleColor}`}>{data.content}</p>
+
+      <div className="custom-hr"></div>
+      <div className="priority">
+        Priority Level:{" "}
```
Comment (PuneetP16) [inline src/Components/Cards/Cards.jsx:66]: remove empty jsx strings

File: src/Components/Filter/Filter.jsx [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+            name="priority"
+            id="high-priority"
+            value={"High"}
+            onChange={(e) =>
+              filterDispatch({
+                type: "PRIORITY_HIGH",
+                payload: e.target.value,
+              })
+            }
+            checked={filter.priorityFilter === "High" ? true : false}
```
Comment (PuneetP16) [inline src/Components/Filter/Filter.jsx:76]: Suggestion: no need to use ternary operator here, as the test condition itself will return either true or false.

File: src/Components/Filter/Filter.jsx [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+            name="priority"
+            id="med-priority"
+            value={"Medium"}
+            onChange={(e) =>
+              filterDispatch({
+                type: "PRIORITY_MED",
+                payload: e.target.value,
+              })
+            }
+            checked={filter.priorityFilter === "Medium" ? true : false}
```
Comment (PuneetP16) [inline src/Components/Filter/Filter.jsx:94]: Here aswell.

File: src/Components/Sidebar/Sidebar.jsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
+import * as All from "../../icons/icons";
+export function Sidebar() {
+  return (
+    <section className="sidebar">
+      <ul className="sidebar-container">
+        <a href="">
```
Comment (PuneetP16) [inline src/Components/Sidebar/Sidebar.jsx:6]: Suggestion: Use <Link> tag instead of Anchor tag

File: src/Components/SignIn/SignIn.css [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+  row-gap: 9px;
+}
+#test-login {
+  grid-area: test;
+}
+.form-label {
+  color: var(--white);
+}
+
+#email-label {
```
Comment (PuneetP16) [inline src/Components/SignIn/SignIn.css:20]: Suggestion: give consistent spacing between css declarations.

File: src/Components/SignIn/SignIn.css [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+}
+#form-btn2 {
+  grid-area: btn2;
+  text-align: center;
+  font-size: 13.33px;
+}
+
+.form-container {
+  display: flex;
+  place-content: center;
```
Comment (PuneetP16) [inline src/Components/SignIn/SignIn.css:45]: Suggestion: Either use class selectors or ID selectors everytime to keep the code consistent.

File: src/Func/NoteAction.jsx [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+        ...note,
+        label: action.payload,
+      };
+    case "REMOVE_LABEL":
+      return {
+        ...note,
+        label: note.label.filter((label) => label !== action.removeLabel),
+      };
+    case "EDIT":
+      console.log("editDATA", action.editData);
```
Comment (PuneetP16) [inline src/Func/NoteAction.jsx:25]: suggestion: remove console.log()

File: src/Pages/Home/Home.jsx [hunk 11]
```diff
@@ -0,0 +1,276 @@
...
+            url: `/api/notes`,
+            data: { note: note },
+            headers: { authorization: token },
+          });
+          //   setNotesData([...response.data.notes]);
+          setOthersData(
+            response.data.notes.filter((note) => note.pinned === false)
+          );
+          //   setPinnedData(response.data.notes.filter((note) => note.pinned));
+
```
Comment (PuneetP16) [inline src/Pages/Home/Home.jsx:64]: remove commented code if no longer required

**Review by PuneetP16 — APPROVED** [review]
Great work on functionality. left few suggestion to work on. good to merge.

File: src/Pages/Home/Home.jsx [hunk 12]
```diff
@@ -0,0 +1,276 @@
+import axios from "axios";
+import { useState } from "react";
+import { useEffect } from "react";
+import { Card } from "../../Components/Cards/Cards";
+import { Filter } from "../../Components/Filter/Filter";
```
Comment (pankaj-prog) [inline src/Pages/Home/Home.jsx:5]: suggestion: you can create a single file for exporting components and import all ur components from that file

File: src/Pages/Home/Home.jsx [hunk 13]
```diff
@@ -0,0 +1,276 @@
...
+    } else {
+      editNoteHandler(note._id);
+    }
+  };
+
+  async function editNoteHandler(notesId) {
+    try {
+      const response = await axios({
+        method: "POST",
+        url: `/api/notes/${notesId}`,
```
Comment (pankaj-prog) [inline src/Pages/Home/Home.jsx:79]: suggestion: try implementing making custom Axios hook

File: src/Pages/Home/Home.jsx [hunk 14]
```diff
@@ -0,0 +1,276 @@
...
+    dispatch({
+      type: "PRIORITY",
+      setPriority: e.target.value,
+    });
+  };
+
+  let filteredData = priority(othersData, filter.priorityFilter);
+  filteredData = labelFilter(filteredData, filter.labelFilter);
+  filteredData = SortByDate(filteredData, filter.sortByDate);
+  console.log("test filter", othersData);
```
Comment (pankaj-prog) [inline src/Pages/Home/Home.jsx:114]: remove console's

**Review by pankaj-prog — APPROVED** [review]

**Pull request merged**